dev/tools/localization/gen_l10n.dart

```diff
@@ -473,13 +467,76 @@ Future<void> main(List<String> arguments) async {
     'pub',
     'run',
     'intl_translation:generate_from_arb',
-    '--output-dir=${l10nDirectory.path}',
+    '--output-dir=${localizationsGenerator.l10nDirectory.path}',
     '--no-use-deferred-loading',
-    outputFile.path,
+    localizationsGenerator.outputFile.path,
     ...arbFilenames,
   ]);
   if (generateFromArbResult.exitCode != 0) {
     stderr.write(generateFromArbResult.stderr);
     exit(1);
   }
 }
+
+class LocalizationsGenerator {
+  LocalizationsGenerator(
+    this._fs,
+    String arbPathString,
+    String templateArbFileName,
+    String outputFileString,
+    String classNameString,
+  ) {
+    _setL10nDirectory(arbPathString);
+    _setTemplateArbFile(templateArbFileName);
+    _setOutputFile(outputFileString);
+    _setClassName(classNameString);
+  }
+
+  final local.LocalFileSystem _fs;
+
+  /// TODO: documentation
+  Directory l10nDirectory;
+
+  /// TODO: documentation
+  File templateArbFile;
+
+  /// TODO: documentation
+  File outputFile;
+
+  void _setL10nDirectory(String arbPathString) {
```

Comment:
Since these setters are done only once, would it make sense to remove these methods and have all of the logic in the constructor? I was thinking of it more so from an organization of code perspective, but having them as private methods when they're only used once doesn't quite make sense either.